# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../rezound/gcc-4.0-64.patch
# Copyright (C) 2004 - 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Gcc-4.0 does not like loosing precision in pointer casts.

  - Rene Rebe <rene@exactcode.de>

--- rezound-0.12.2beta/src/frontend_fox/CPasteChannelsDialog.cpp.vanilla	2005-10-28 12:01:58.000000000 +0200
+++ rezound-0.12.2beta/src/frontend_fox/CPasteChannelsDialog.cpp	2005-10-28 12:03:12.000000000 +0200
@@ -206,7 +206,7 @@
 	{
 		pasteChannels.clear();
 
-		actionParameters->setValue<unsigned>(_("MixMethod"),(unsigned)(mixTypeComboBox->getItemData(mixTypeComboBox->getCurrentItem())));
+		actionParameters->setValue<unsigned>(_("MixMethod"),(long)(mixTypeComboBox->getItemData(mixTypeComboBox->getCurrentItem())));
 		
 		if(repeatTypeComboBox->getCurrentItem()==0)
 		{ // repeating it a given number of times
--- rezound-0.12.2beta/src/frontend_fox/CMainWindow.cpp.vanilla	2005-10-28 11:55:32.000000000 +0200
+++ rezound-0.12.2beta/src/frontend_fox/CMainWindow.cpp	2005-10-28 12:01:13.000000000 +0200
@@ -433,7 +433,7 @@
 
 long CMainWindow::onSoundListChange(FXObject *sender,FXSelector sel,void *ptr)
 {
-	FXint index=(FXint)ptr;
+	FXint index=(long)ptr;
 
 	if(index>=0 && index<soundList->getNumItems())
 	{
@@ -1836,13 +1836,13 @@
 
 	if(pos==0 && FXSELID(sel)==ID_SHUTTLE_BACKWARD)
 	{
-		if((unsigned)ptr > 1/*came from keyboard event*/) enableAutoRepeat(getApp()->getDisplay(),false); // QQQ
+		if((long)ptr > 1/*came from keyboard event*/) enableAutoRepeat(getApp()->getDisplay(),false); // QQQ
 		shuttleDial->setValue(pos-inc);
 		onShuttleChange(sender,sel,ptr);
 	}
 	else if(pos==0 && FXSELID(sel)==ID_SHUTTLE_FORWARD)
 	{
-		if((unsigned)ptr > 1/*came from keyboard event*/) enableAutoRepeat(getApp()->getDisplay(),false); // QQQ
+		if((long)ptr > 1/*came from keyboard event*/) enableAutoRepeat(getApp()->getDisplay(),false); // QQQ
 		shuttleDial->setValue(pos+inc);
 		onShuttleChange(sender,sel,ptr);
 	}
--- rezound-0.12.2beta/src/frontend_fox/COggDialog.cpp.vanilla	2005-10-28 12:04:39.000000000 +0200
+++ rezound-0.12.2beta/src/frontend_fox/COggDialog.cpp	2005-10-28 12:04:44.000000000 +0200
@@ -136,7 +136,7 @@
 
 long COggDialog::onRadioButton(FXObject *sender,FXSelector sel,void *ptr)
 {
-	if((int)ptr==0) // only act when ptr==1 when it's getting checked
+	if((long)ptr==0) // only act when ptr==1 when it's getting checked
 		return 1;
 
 	// turn off all buttons
--- rezound-0.12.2beta/src/frontend_fox/CMp3Dialog.cpp.vanilla	2005-10-28 12:05:01.000000000 +0200
+++ rezound-0.12.2beta/src/frontend_fox/CMp3Dialog.cpp	2005-10-28 12:05:05.000000000 +0200
@@ -170,7 +170,7 @@
 
 long CMp3Dialog::onRadioButton(FXObject *sender,FXSelector sel,void *ptr)
 {
-	if((int)ptr==0) // only act when ptr==1 when it's getting checked
+	if((long)ptr==0) // only act when ptr==1 when it's getting checked
 		return 1;
 
 	// turn off all buttons
